src/braket/circuits/circuit.py

```diff
@@ -436,26 +438,36 @@ def _check_if_qubit_measured(
                 qubit mappings to apply to the `instruction.target`. Key is the qubit in
                 `instruction.target` and the value is what the key will be changed to.
                 Default = `None`.
+            target_qubits (QubitSet | None): the target qubits for a Noise `instruction`.
+                Default = `None`.
 
         Raises:
             ValueError: If adding a gate or noise operation after a measure instruction.
         """
-        if (
-            # check if there is a measure instruction on the target qubit
-            target
-            and target in self._measure_targets
-            # check if there is a measure instruction on any qubits in the target_mapping
-            or (target_mapping and any(targ in self._measure_targets for targ in target_mapping))
-            # If no target or target_mapping is supplied, check if there is a measure
-            # instruction on the current instructions target qubit
-            or (
-                instruction.target
-                and any(targ in self._measure_targets for targ in instruction.target)
-            )
-        ):
-            raise ValueError(
-                "cannot add a gate or noise operation on a qubit after a measure instruction."
-            )
+        if self._measure_targets:
+            if (
+                # check if the target qubits are in the measured targets
+                target_qubits
+                and any(target_qubits in self._measure_targets)
+                or (
+                    # check if there is a measure instruction on the target qubit
+                    target
+                    and target in self._measure_targets
+                )
+                # check if there is a measure instruction on any qubits in the target_mapping
+                or (
+                    target_mapping and any(targ in self._measure_targets for targ in target_mapping)
+                )
+                # If no target or target_mapping is supplied, check if there is a measure
+                # instruction on the current instructions target qubit
```

Comment:
these conditions are extremely difficult to read. Could you refactor this to something simpler? Perhaps 
```
cond_1 = A and B
cond_2 = C  and D
...
if cond_1 or cond_2: ... 
```
could work

Reply:
Refactored these as you suggested.